### PR TITLE
[Documentation] Concepts: Currencies

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | yes
| New docs?    | yes
| BC breaks?      | no
| Related tickets | partially #5275 
| License         | MIT

### CODE REVIEW

**Comment by pamil** [comment]
Will be quite outdated after #5497 :tada:

**Comment by TheMadeleine** [comment]
@pamil not everything, but yes. Some parts are waiting for its merge :)

**Comment by pjedrzejewski** [comment]
@TheMadeleine I think we are missing example of switching the currency, hm?

**Comment by pamil** [comment]
There's `sylius.storage.currency` service implementing `Sylius\Component\Core\Currency\CurrencyStorageInterface` with methods `set(ChannelInterface $channel, $currencyCode)` and `get(ChannelInterface $channel)`.

So `$container->get('sylius.storage.currency')->set($channel, 'PLN')` may be an example.

**Comment by pjedrzejewski** [comment]
@pamil What do you think about switching the currency on the context itself? That would be of course separate interface and implementation. Because using storage for that sounds weird. Storage should probably be a private service.

**Comment by pamil** [comment]
This would require additional type checks before any call to that context. Moreover, creating decorators for contexts would be nearly impossible. 

Let's say we've got `SettableCurrencyContextInterface`, so decorator for it should also implement the same. But since we can also decorate a regular context, we need to perform a type check inside `set()` in the decorator, which results that we can't rely on that interface fully.

I don't like those storage mess too much, but it seems much simplier for now.

File: docs/book/currencies.rst [hunk 1]
```diff
@@ -7,57 +7,59 @@ Currencies
...
+        // or the other way - convert 100 PLN to amount in the base currency
+        $this->get('sylius.currency_converter')->convertToBase(100, 'PLN');
+    }
 
 Available Currencies Provider
 -----------------------------
 
-The default menu for selecting currency is using a special service called ``sylius.currency_provider``, which returns all enabled currencies.
+The default menu for selecting currency is using a service - **CurrencyProvider** - with the ``sylius.currency_provider`` id, which returns all enabled currencies.
 This is your entry point if you would like override this logic and return different currencies for various scenarios.
```
Comment (NeverResponse) [inline docs/book/currencies.rst]: `to override`?

**Comment by pjedrzejewski** [comment]
Thanks Magda and Kamil!

**Pull request merged**